enhancements/installer/cluster-api-based-installations.md

```diff
@@ -0,0 +1,347 @@
+
+- Non-goal: To maintain a consistent infrastructure footprint as clusters previously created with Terraform
+- Future work: To optimize build processes or binary size (this will be prioritized ASAP in a separate design doc)
+- Future work: To use an existing management cluster to install OpenShift
+- Future work: To pivot the CAPI manifests to the newly-installed cluster to enable day-2 infrastructure management within the cluster.
+
+## Proposal
+
+The Installer will create CAPI infrastructure manifests based on user
+input from the install config; then, in order to provision cluster infrastructure,
+apply the manifests to CAPI controllers running on a local Kubernetes control-plane
+setup by [envtest](https://pkg.go.dev/sigs.k8s.io/controller-runtime/pkg/envtest). 
+
+### Workflow Description
+
+**cluster creator** is a human user responsible for deploying a
+cluster. Note that the workflow does not change for this user.
+
+**openshift-install** is the Installer binary.
+
+1. The cluster creator provides an install-config and credentials
+2. (optional) The cluster creator runs `openshift-install create manifests`
+3. (optional) The cluster creator edits the newly created CAPI manifests.
+4. The cluster creator runs `openshift-install create cluster`
+5. `openshift-install` extracts binaries for kube-apiserver, etcd, CAPI infrastructure provider & cloud CAPI provider to the install dir
+6. `openshift-install` using `envtest` initializes a control plane locally on the Installer host
+7. `openshift-install` execs the CAPI infrastructure and cloud provider as subprocesses, pointing them to the local control plane
+8. `openshift-install` applies the CAPI manifests to the control plane
+9. The CAPI controllers provision cluster infrastructure based on the manifests
+10. `openshift-install` monitors the status of the local manifests as they are applied
+11. If the statuses are as expected, infrastrucutre has been provisioned and installation continues with the normal flow.
+
+In the case of an error in the final step, the Installer will bubble up resources with non-expected statuses.
+
+#### Variation and form factor considerations [optional]
+
+
+### API Extensions
+
+As a result of this enhancement, API Extensions will only be used locally by the 
+local Installer control plane--they will not, until future work, be pivoted to the cluster.
+The following are CRD references for Installer-supported platforms:
+
+- AWS: [infrastructure.cluster.x-k8s.io/v1beta2][awsCRD]
+- Azure/AzureStack: [infrastructure.cluster.x-k8s.io/v1beta1][azureCRD]
+- GCP: [infrastructure.cluster.x-k8s.io/v1beta1][gcpCRD]
+- IBMCloud/PowerVS: [infrastructure.cluster.x-k8s.io/v1beta2][ibmCRD]
+- Nutanix: [infrastructure.cluster.x-k8s.io/v1beta1][nutanixCRD]
+- OpenStack: [infrastructure.cluster.x-k8s.io/v1alpha7][openstackCRD]
+- vSphere: [infrastructure.cluster.x-k8s.io/v1beta1][vsphereCRD]
+
+### Implementation Details/Notes/Constraints [optional]
```

Comment:
What did we settle on for how to handle creation and deletion of the bootstrap host that openshift uses during installation? I assume most CAPI providers don't have that as a pattern.

Reply:
This is a good question. This doc is focused on how the installer will interact with CAPI. Is there a separate design doc for how we get the CAPI component(s) we need to actually produce a cluster?

Reply:
> What did we settle on for how to handle creation and deletion of the bootstrap host that openshift uses during installation? I assume most CAPI providers don't have that as a pattern.

Assuming we're referring to the boostrap machine, the installer flow today still takes into account the creation of a bootstrap machine as a Cluster API Machine object, which is then deleted through a delete call to the api-server (like one would delete any other Cluster API machine).

Given that this flow doesn't reuse any of the `kubeadm` upstream provide assumptions or limitations, we're able to create an OpenShift cluster under Cluster API using primitive APIs (Machine and their related ignition secrets).

> This is a good question. This doc is focused on how the installer will interact with CAPI. Is there a separate design doc for how we get the CAPI component(s) we need to actually produce a cluster?

Good point, I'll take an action item to document the current implementation in https://github.com/openshift/installer/blob/master/docs/dev/cluster-api.md which can be useful for future implementers.